core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt

```diff
@@ -1346,9 +1353,53 @@ abstract class CoreReaderFragment :
           ).show()
         }
       }
+      REQUEST_POST_NOTIFICATION_PERMISSION -> {
+        if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+          onReadAloudMenuClicked()
+        }
+      }
     }
   }
 
+  @RequiresApi(Build.VERSION_CODES.TIRAMISU)
+  private fun hasNotificationPermission() =
+    ContextCompat.checkSelfPermission(
+      requireActivity(),
+      POST_NOTIFICATIONS
+    ) == PackageManager.PERMISSION_GRANTED
+
+  private fun checkNotificationPermission(): Boolean {
+    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+      hasNotificationPermission().also { permissionGranted ->
+        if (!permissionGranted) {
+          if (ActivityCompat.shouldShowRequestPermissionRationale(
```

Comment:
> If the device is lower than 12 it'll show once

@gouri-panda , this permission is only for android 13 and above, so for below android 12 it'll not working.

> What happens when the user didn't give the permission first time. We should make that case as well.

I'm refactoring the code for this.

Reply:
if the user does not give permission first time then it will ask again (when user click on zim for downloading zim file , when click on read aloud, when click on start server) and if user denied for first time then now it will show  permission again and if user again denied for permission then it will show permission dialog.